### ISSUE TITLE
Chinese character date translation format problem

### ISSUE THREAD

**Issue by DaPoHou** [fisharebest/webtrees#4253]
'年' is year，’月' is month，'日' is day
When I translated Chinese, I found that when the space is not reserved before %n
  `'%j %F %Y' => '%Y年%n月%j日',`

 The code will be wrong,add a word '月' below the input box
![dateerror](https://user-images.githubusercontent.com/26761596/156957054-ffa9a6bc-c885-42ee-8206-09ff289c1927.png)

The code here enters an infinite loop:
https://github.com/fisharebest/webtrees/blob/718572693f6a54a91ebf59f26621acfb21b425d3/app/Date.php#L179

If I don't use Chinese characters between %Y, %n, %j, everything is normal.
  `'%j %F %Y' => '%Y-%n-%j-',`

**Comment by fisharebest** [comment]
The translation for chinese is `'%j %F %Y' => '%Y年 %n月 %j日'`

Do you use this, or have you changed it.  e.g. `'%j %F %Y' => '%Y年,%n月,%j日'`

**Comment by DaPoHou** [comment]
> The translation for chinese is `'%j %F %Y' => '%Y年 %n月 %j日'`
> 
> Do you use this, or have you changed it. e.g. `'%j %F %Y' => '%Y年,%n月,%j日'`

Yes, I strip out the spaces between characters and it produces the error.
`'%j %F %Y' => '%Y年%n月%j日'`

**Comment by fisharebest** [comment]
When a date is incomplete (e.g. just a year), then we must remove extra characters.

For example, American-English uses `'%j %F %Y' => '%F %j, %Y'`.

With just a year, we would have `  , 2000`.  Therefore we must remove the extra characters.

e.g. https://github.com/fisharebest/webtrees/blob/2.0.19/app/Date/AbstractCalendarDate.php#L570-L578

For Chinese, we remove `'年 %n月'`.

This is the text that we expect.

You changed the translation.  Which is correct.  With spaces?  Without spaces?

**Comment by DaPoHou** [comment]
There is no need for spaces when Chinese and Arabic numerals are identified together.(Usually, spaces are not used between Chinese characters and other characters.）

When there is only a year, we would have `2000年`
If there is a month,  we would  have `2000年3月`

**Comment by fisharebest** [comment]
> When there is only a year, we would have 2000年

Another Chinese user told me that if there is only a year, you must write `2000` - not `2000年`.  Is this correct?

**Comment by DaPoHou** [comment]
> 

This is wrong. If it is only  2000 and there is no `年` in the date representation, The user has no idea what its unit is.
For example, there is only one number 12, not have `年`,So does this mark the month or year?Or is it age?Users will feel confused.

**Comment by DaPoHou** [comment]
There is a case where the `年` can be omitted.
for example 1985－1987
